components/brave_rewards/browser/rewards_service_browsertest.cc

```diff
@@ -461,3 +976,228 @@ IN_PROC_BROWSER_TEST_F(BraveRewardsBrowserTest, HandleFlagsWrongInput) {
+  ui_test_utils::NavigateToURLWithDisposition(
+      browser(), url, WindowOpenDisposition::NEW_FOREGROUND_TAB,
+      ui_test_utils::BROWSER_TEST_WAIT_FOR_NAVIGATION);
+
+  // Open the Rewards popup
+  content::WebContents* popup_contents = OpenRewardsPopup();
+  ASSERT_TRUE(popup_contents);
+
+  // Retrieve the inner text of the wallet panel and verify that it
+  // looks as expected
+  content::EvalJsResult js_result = EvalJs(
+      popup_contents,
+      "const delay = t => new Promise(resolve => setTimeout(resolve, t));"
+      "delay(0).then(() => "
+      "  document.querySelector(\"[id='wallet-panel']\").innerText);",
+      content::EXECUTE_SCRIPT_DEFAULT_OPTIONS,
+      content::ISOLATED_WORLD_ID_CONTENT_END);
+  EXPECT_NE(js_result.ExtractString().find("Brave Verified Publisher"),
+            std::string::npos);
+
+  // Stop observing the Rewards service
+  rewards_service_->RemoveObserver(this);
+}
+
+// #4a - Visit verified publisher
+IN_PROC_BROWSER_TEST_F(BraveRewardsBrowserTest, VisitVerifiedPublisher) {
+  // Observe the Rewards service
+  rewards_service_->AddObserver(this);
+
+  // Enable Rewards
+  EnableRewards();
+
+  // Visit verified publisher
+  bool verified = true;
+  VisitPublisher("duckduckgo.com", verified);
+
+  // Stop observing the Rewards service
+  rewards_service_->RemoveObserver(this);
+}
+
+// #4b - Visit unverified publisher
+IN_PROC_BROWSER_TEST_F(BraveRewardsBrowserTest, VisitUnverifiedPublisher) {
+  // Observe the Rewards service
+  rewards_service_->AddObserver(this);
+
+  // Enable Rewards
+  EnableRewards();
+
+  // Visit unverified publisher
+  bool verified = false;
+  VisitPublisher("google.com", verified);
+
+  // Stop observing the Rewards service
+  rewards_service_->RemoveObserver(this);
+}
+
+// #5 - Auto contribution
+IN_PROC_BROWSER_TEST_F(BraveRewardsBrowserTest, AutoContribution) {
+  // Observe the Rewards service
+  rewards_service_->AddObserver(this);
+
+  // Enable Rewards
+  EnableRewards();
+
+  // Claim grant using panel
+  bool use_panel = true;
+  ClaimGrant(use_panel);
+
+  // Visit verified publisher
+  bool verified = true;
+  VisitPublisher("duckduckgo.com", verified);
+
+  // Trigger auto contribution now
+  rewards_service()->StartAutoContributeForTest();
+
+  // Signal that donation was made and update wallet with new balance
+  donation_made_ = true;
+  UpdateTestData();
+
+  // Wait for reconciliation to complete successfully
+  WaitForReconcileCompleted();
+
+  // Make sure that balance is updated correctly
+  {
+    content::EvalJsResult js_result = EvalJs(
+        contents(),
+        "const delay = t => new Promise(resolve => setTimeout(resolve, t));"
+        "delay(1000).then(() => "
+        "  "
+        "document.querySelector(\"[data-test-id='balance']\").innerText);",
+        content::EXECUTE_SCRIPT_DEFAULT_OPTIONS,
+        content::ISOLATED_WORLD_ID_CONTENT_END);
+    EXPECT_NE(js_result.ExtractString().find("29.0 BAT"), std::string::npos);
```

Comment:
Default AC amount is 20BAT. As far as I can see this happens, because wallet balance is mocked to 29 BAT so that the same one can be used for tip and ac. Which is ok, but it's a little bit confusing as this doesn't reflect an actual AC amount. Could we maybe have separate response for AC contribution?